src/index.js

```diff
@@ -5,18 +5,23 @@ import bs58 from 'bs58';
 export default class Wallet extends EventEmitter {
   constructor(providerUrl, network) {
     super();
-    this._providerUrl = new URL(providerUrl);
-    this._providerUrl.hash = new URLSearchParams({
-      origin: window.location.origin,
-      network,
-    }).toString();
+    try {
+      this._providerUrl = new URL(providerUrl);
+      this._providerUrl.hash = new URLSearchParams({
+        origin: window.location.origin,
+        network,
+      }).toString();
+      this._useInjectedInterface = false;
+    } catch (err) {
+      // invalid URL, use injected interface
+      this._useInjectedInterface = true;
```

Comment:
what if `useInjectedInterface` was instead a parameter you pass in at construction?

Reply:
yes, it's also a solution, but since we don't use the providerUrl for injected wallet, we might as well as reuse the param, and it much easier to update the dapp code (like serum-dex-ui) by just simply add another wallet: `{name: "injected", url: "-"}`